website/static/js/fileBrowser.js

```diff
@@ -0,0 +1,1175 @@
+/**
+ * Builds full page project browser
+ */
+'use strict';
+
+var $ = require('jquery');  // jQuery
+var m = require('mithril'); // exposes mithril methods, useful for redraw etc.
+var ProjectOrganizer = require('js/project-organizer');
+var $osf = require('js/osfHelpers');
+var LogText = require('js/logTextParser');
+var AddProject = require('js/addProjectPlugin');
+var mC = require('js/mithrilComponents');
+
+if (!window.fileBrowserCounter) {
+    window.fileBrowserCounter = 0;
+}
+function getUID() {
+    window.fileBrowserCounter = window.fileBrowserCounter + 1;
+    return window.fileBrowserCounter;
+}
+
+var xhrconfig = function (xhr) {
+    xhr.withCredentials = true;
+    xhr.setRequestHeader('Content-Type', 'application/vnd.api+json');
+    xhr.setRequestHeader('Accept', 'application/vnd.api+json; ext=bulk');
+};
+
+// Refactor the information needed for filtering rows
+function _formatDataforPO(item) {
+    item.kind = 'folder';
```

Comment:
I see why argument mutation would be a problem but item in this case should have the .kind attribute throughout its existence in this dashboard application. Isn't this a bit overkill for this purpose?